cds_dojson/marc21/fields/books/book.py

```
@@ -0,0 +1,662 @@
+            _acquisition_source.update(
+                {'datetime': datetime.date(year, month, 1).isoformat(),
+                 'source': source or clean_val('9', value, str)})
+        except UnexpectedValue as e:
+            self['_private_notes'] = private_notes(self, key, value)
+            raise IgnoreKey('acquisition_source')
+
+    return _acquisition_source
+
+
+@model.over('_private_notes', '^595__')
+@filter_list_values
+def private_notes(self, key, value):
+    """Translates private notes field."""
+    _priv_notes = self.get('_private_notes', [])
+
+    for v in force_list(value):
+        note = {'value': clean_val('a', v, str, req=True),
+                'source': clean_val('9', v, str),
+                }
+        _priv_notes.append(note)
+    return _priv_notes
+
+
+@model.over('_collections', '(^980__)|(^690C_)|(^697C_)')
+@out_strip
+def collection(self, key, value):
+    """Translates collection field - WARNING - also document type field."""
+    _collections = self.get('_collections', [])
+
+    for v in force_list(value):
+        result_a = mapping(COLLECTION, clean_val('a', v, str))
+        result_b = mapping(COLLECTION, clean_val('b', v, str))
+        if result_a or result_b:
+            _collections.append(result_a)
+            _collections.append(result_b)
+        else:
+            self['document_type'] = document_type(self, key, value)
+            raise IgnoreKey('_collections')
+    return _collections
+
+
+@model.over('document_type', '(^980__)|(^960__)|(^690C_)')
+@out_strip
+def document_type(self, key, value):
+    """Translates document type field."""
+    _doc_type = self.get('document_type', [])
+
+    def doc_type_mapping(val):
+        if val:
+            return mapping(DOCUMENT_TYPE, val)
+
+    for v in force_list(value):
+        _doc_type.append(doc_type_mapping(clean_val('a', v, str)))
+        _doc_type.append(doc_type_mapping(clean_val('b', v, str)))
+    return _doc_type
```

we might have to raise instead of returning `[]` in case the `value` list is empty

---

It's true, because in the data model the document type is required I thought that every record in legacy has it, but as I am now running the reports I can see that is not the case. I'll add the exception